src/test/java/io/vertx/test/core/TimerTest.java

```diff
@@ -163,26 +164,33 @@ public void testTimerStreamExceptionDuringHandle() throws Exception {
   @Test
   public void testTimerStreamCallingWithNullHandlerCancelsTheTimer() throws Exception {
     ReadStream<Long> timer = vertx.timerStream(10);
+    AtomicInteger count = new AtomicInteger();
     timer.handler(l -> {
-      fail();
-    });
-    timer.endHandler(v -> {
-      testComplete();
+      if (count.incrementAndGet() == 1) {
+        timer.handler(null);
+        vertx.setTimer(200, id -> {
+          assertEquals(1, count.get());
+          testComplete();
+        });
+      } else {
+        fail();
+      }
     });
-    timer.handler(null);
     await();
   }
 
   @Test
   public void testTimerStreamCancellation() throws Exception {
     TimeoutStream timer = vertx.timerStream(10);
```

Comment:
I think I can improve the test by instead wrapping it in a runOnContext, so we are sure that it will never be executed.